### PR TITLE
text update

### LINKED ISSUE #409: Font and text are too interlinked
for example, changing the text.font_size also changes the font.size which changes all the other text's font sizes as well.

A temp fix is to clone the font every time but that's expensive and defeats the point of font.

### PR DESCRIPTION
**Pull request by mrElnekave**
# Description

fixes #409 
## Hello reviewers!
Before approving this PR, please ensure the following:

  - [ ] Documentation is up to date
  - [ ] Relevent tutorials are up to date
  - [ ] Changelog is up to date
  - [ ] That the new code actually works

Once all this is done, you can be sure that the PR is ready to be merged. 🎉

*This message has been added automatically. Please don't delete it.*
## Hello reviewers!
Before approving this PR, please ensure the following:

  - [ ] Documentation is up to date
  - [ ] Relevent tutorials are up to date
  - [ ] Changelog is up to date
  - [ ] That the new code actually works

Once all this is done, you can be sure that the PR is ready to be merged. 🎉

*This message has been added automatically. Please don't delete it.*

### CODE REVIEW

File: rubato/structure/gameobject/ui/text.py [hunk 1]
```diff
@@ -4,12 +4,13 @@
...
 from .. import Component
-from .... import Vector, Color, Font, Draw, Camera, Surface
+from .... import Vector, Font, Draw, Camera, Surface
 
 
 class Text(Component):
     """
-    A text component. Add this to game objects or UI elements to give them text.
+    A text component. Add this to game objects or UI elements to give them text. Takes a font object to render the text.
+    Any change to the characters of the text need to be done directly to the font object.
```
Comment (tinmarr) [inline rubato/structure/gameobject/ui/text.py:13]: delete line 13

**Review by mrElnekave — COMMENTED** [review]
Ready for review, looks good

File: CHANGELOG.md [hunk 2]
```diff
@@ -28,6 +30,8 @@
 
 -   The default window size now matches the resolution instead of being half it to be more intuitive.
 -   Changed default resolution to (500, 500).
+-   Font add_style and remove_style no longer look if the thing you are removing or adding exists.
+-   Font styles are integers you | together.
```
Comment (tinmarr) [inline CHANGELOG.md:34]:
```suggestion
-   `font.add_style` and `font.remove_style` no longer look if the thing you are removing or adding exists.
-   `Font` styles are integers you | together.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -20,6 +20,8 @@
 
 ### Breaking Changes
 
+-   Text no longer can affect Fonts directly.
```
Comment (tinmarr) [inline CHANGELOG.md:23]:
```suggestion
-   `Text` no longer can affect `Font`s directly.
```

File: demo/test/text.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import rubato as rb
```
Comment (tinmarr) [inline demo/test/text.py:1]: Please delete this file. We don't want a test folder in the demo folder anymore

File: rubato/utils/rendering/font.py [hunk 5]
```diff
@@ -15,8 +15,8 @@ class Font:
         font: The font to use. Can also be a path to a font file. Defaults to Roboto.
             Included fonts are "Comfortaa", "Fredoka", "Merriweather", "Roboto", "SourceCodePro", "Mozart"
         size: The size of the font in pixels. Defaults to 16.
-        styles: The styles to apply to the font. Defaults to [].
-            Fill with only the following: bold, italic, underline, strikethrough.
+        styles: The styles to apply to the font. Defaults to 0. You can add multiple styles by | them together.
```
Comment (tinmarr) [inline rubato/utils/rendering/font.py:18]:
```suggestion
        styles: The styles to apply to the font. Set multiple styles by adding them together. Defaults to 0.
```

File: rubato/utils/rendering/font.py [hunk 6]
```diff
@@ -117,39 +116,38 @@ def size_text(self, text: str) -> tuple[int, int]:
...
 
-    def add_style(self, style: str):
+    def add_style(self, style: int):
         """
         Adds a style to the font.
 
         Args:
-            style: The style to add. Can be one of the following: bold, italic, underline, strikethrough.
+            style: The style to add. You can add multiple styles by | them together.
+            Can be one of the following: Font.BOLD, Font.ITALIC, Font.UNDERLINE, Font.STRIKETHROUGH.
```
Comment (tinmarr) [inline rubato/utils/rendering/font.py:125]: Move line 125 to the function description (inline with line 121)

File: rubato/utils/rendering/font.py [hunk 7]
```diff
@@ -117,39 +116,38 @@ def size_text(self, text: str) -> tuple[int, int]:
...
 
-    def remove_style(self, style: str):
+    def remove_style(self, style: int):
         """
         Removes a style from the font.
 
         Args:
-            style: The style to remove. Can be one of the following: bold, italic, underline, strikethrough.
+            style: The style to remove.
+            Can be one of the following: Font.BOLD, Font.ITALIC, Font.UNDERLINE, Font.STRIKETHROUGH.
```
Comment (tinmarr) [inline rubato/utils/rendering/font.py:140]: Same as previous comment

**Review by tinmarr — CHANGES_REQUESTED** [review]

**Review by tinmarr — APPROVED** [review]

**Pull request merged**